pkg/apis/configuration/v1alpha1/ingress_class_param_types.go

```diff
@@ -38,6 +38,7 @@ type IngressClassParametersList struct {
 //+kubebuilder:object:root=true
 //+kubebuilder:storageversion
 //+kubebuilder:resource:categories=kong-ingress-controller
+//+kubebuilder:resource:path=ingressclassparameterses
```

Comment:
How about considering to change this to `//+kubebuilder:resource:path=ingressclassparameters` to be compatible to previous versions, and `ingressparameters` is already a plural form

Reply:
It is a linguistic plural form, but as far as the CRD is concerned that plural is the singular form. Were they the same, there'd be no way to distinguish the two. I don't think this strictly matters (I don't think kubectl distinguishes either--you can use the plural to get a single resource for other kinds), but it works if you treat the first form as a singular form--an "IngressClassParameters" is a single CR that contains a list of ingress class parameters, and it follows the normal "-es" suffix pattern to create a plural for words that already end in "s", like "octopuses" (ignore that there are two other valid plural forms of "octopus").

As a more practical matter, that's the form client-gen wants to use. I could find a way to override kubebuilder's defaults easily, client-gen not so much. If both are valid, I'll take the one that's easier to do.